source/extensions/tracers/zipkin/zipkin_tracer_impl.h

```diff
@@ -80,6 +80,9 @@ class ZipkinSpan : public Tracing::Span {
   void setBaggage(absl::string_view, absl::string_view) override;
   std::string getBaggage(absl::string_view) override;
 
+  // TODO: This method is unimplemented for Zipkin.
+  std::string getTraceId() const override { return std::string(); };
```

Comment:
FYI zipkin might have wanted the same thing and exposed this in their span subclass and use a cast on the return value of `startSpan` to access it.
https://github.com/envoyproxy/envoy/blob/ee25537fa6915ce081d46cfa7f6059630633696f/source/extensions/tracers/zipkin/zipkin_core_types.h#L523

Out of curiousity, did you need to expose this for all types of spans?

Reply:
In our custom filter, we would like to access the trace ID for a span created by HTTP Connection Manager. Ideally, we want this to work no matter which tracer is used. 

IMO casting to a specific span subclass seems like an anti-pattern: we need to add an explicit dependency from our filter to this tracer. And we plan to switch to OpenTelemetry in the future, so this would make it cleaner.

Reply:
Gotcha, thanks for helping me understand the use case!